### DISCUSSION TITLE
Using Stumpy to find general shape within time series?

### DISCUSSION THREAD

**Discussion by forrestmckee** [TDAmeritrade/stumpy#336] [Q&A]
Using Stumpy, is it possible to detect a general pattern within a time series? For simplicity, I'm looking to detect an on/off "step," (think square wave) but don't know how many time steps it will take for that pattern to happen. The goal would be to pass in two time series. One is a general pattern, the other is the time series I'm looking in (I know 2 time series can be passed in similar to the ice-ice baby example in the docs). The problem I'm running into is the length or magnitude of the pattern could be significantly smaller than the time series, but the overall pattern could be similar. Is there a way to do this using stumpy? Maybe some sort of normalization?

Please let me know if my question is unclear and I'll try and rephrase.

**Comment by seanlaw** [comment]
@fgmckee95 I think what you might be looking for is related to [this SDTS paper](https://www.cs.ucr.edu/~eamonn/WeaklyLabeledTimeSeries.pdf). Unfortunately, we don't have a tutorial on this yet.
